lib/topologies/mongos.js

```diff
@@ -879,18 +882,39 @@ Mongos.prototype.isDestroyed = function() {
 
 // Execute write operation
 var executeWriteOperation = function(self, op, ns, ops, options, callback) {
-  if (typeof options === 'function') {
-    (callback = options), (options = {}), (options = options || {});
-  }
-
-  // Ensure we have no options
+  if (typeof options === 'function') (callback = options), (options = {});
   options = options || {};
+
   // Pick a server
-  var server = pickProxy(self);
+  let server = pickProxy(self);
   // No server found error out
   if (!server) return callback(new MongoError('no mongos proxy available'));
-  // Execute the command
-  server[op](ns, ops, options, callback);
+
+  if (!options.retryWrites || !options.session || !isRetryableWritesSupported(self)) {
```

Comment:
Should we DeMorgan this?

Reply:
I know its slightly harder to read this way, but imho it actually makes the code much more readable - basically what's being done here is "if none of this is set, then run this minimal piece of code and exit out, otherwise do this much longer/more involved segment."  It also reduces the brace level